lib/active_merchant/billing/gateways/litle.rb

```diff
@@ -223,6 +223,55 @@ def add_auth_purchase_params(doc, money, payment_method, options)
         add_descriptor(doc, options)
         add_merchant_data(doc, options)
         add_debt_repayment(doc, options)
+        add_stored_credential_params(doc, options)
+      end
+
+      def add_stored_credential_params(doc, options={})
+        return unless options[:stored_credential]
+
+        if options[:stored_credential][:initial_transaction]
+          add_stored_credential_params_initial(doc, options)
+        else
+          add_stored_credential_params_used(doc, options)
+        end
+      end
+
+      def add_stored_credential_params_initial(doc, options)
+        case options[:stored_credential][:reason_type]
+        when 'unscheduled'
+          doc.processingType("initialCOF")
+        when 'installment'
+          doc.processingType("initialInstallment")
+        when 'recurring'
+          doc.processingType("initialRecurring")
+        end
+      end
+
+      def add_stored_credential_params_used(doc, options)
+        network_transaction_id = options[:stored_credential].fetch(:network_transaction_id)
```

Comment:
I have not often used `fetch`, is it particularly useful here?

Reply:
Yah, this was one of those soft-blocker questions I alluded to yesterday. What `Hash#fetch` does is get the value at the given key (first argument to `#fetch`), and if that key wasn't found raise an exception. My wanting to explicitly not support calling this method without providing the `:network_transaction_id` is why I used `fetch` here. (This method only gets called if the user is asking to use existing stored credentials, and if they're doing that they should have a network transaction id).

(As a side-note, `#fetch` is also handy if you do want the key to be optional but want to explicitly set the default value to something other than the Hash instance's default when the key isn't provided; I didn't want that here so I didn't pass the second argument to `#fetch`).

BUT, I'm not actually sure if we want to raise an exception here or not. From the perspective of the user of AM, should we alert the user immediately if they're not passing the needed data? Also, if we do still want to raise here, what exception would be appropriate to raise? (Something specific to ActiveMerchant?)

Reply:
Very good questions that I'm not sure I have an easy answer for. Generally, we have leaned away from raising exceptions and toward allowing the gateway to provide whatever error they throw after receiving the request. But this might be a good candidate for standardization along with normalizing stored credentials. `ArgumentError` might be the closest fit.

Reply:
I'll go ahead and not raise in this case, to better match what we're already doing. I don't want to break with convention/consistency needlessly. Thanks for your feedback, it was very helpful!